policy/release/slsa_source_version_controlled.rego

```diff
@@ -0,0 +1,85 @@
+#
+# METADATA
+# title: SLSA - Source - Version Controlled
+# description: |-
+#   The SLSA requirement states the following:
+#
+#   "Every change to the source is tracked in a version control system
+#   that meets the following requirements:
+#
+#     [Change history] There exists a record of the history of changes
+#     that went into the revision. Each change must contain: the
+#     identities of the uploader and reviewers (if any), timestamps of
+#     the reviews (if any) and submission, the change
+#     description/justification, the content of the change, and the
+#     parent revisions.
+#
+#     [Immutable reference] There exists a way to indefinitely reference
+#     this particular, immutable revision. In git, this is the {repo URL +
+#     branch/tag/ref + commit ID}.
+#
+#   Most popular version control system meet this requirement, such as git,
+#   Mercurial, Subversion, or Perforce."
+#
+#   This package verifies the requirement by asserting the image was built
+#   from a git repository.
+#
+package policy.release.slsa_source_version_controlled
+
+import future.keywords.contains
+import future.keywords.if
+import future.keywords.in
+
+import data.lib
+
+# METADATA
+# title: Material format
+# description: |-
+#   At least one entry in the predicate.materials array of the attestation contains
+#   the expected attributes: uri and digest.sha1.
+# custom:
+#   short_name: missing_materials
+#   failure_msg: No materials match expected format
+#
+deny contains result if {
+	count(lib.pipelinerun_attestations) > 0
+	count(materials) == 0
+	result := lib.result_helper(rego.metadata.chain(), [])
+}
+
+# METADATA
+# title: Material from a git repository
+# description: |-
+#   Each entry in the predicate.materials array of the attestation uses
+#   a git URI.
+# custom:
+#   short_name: material_non_git_uri
+#   failure_msg: Material URI %q is not a git URI
+#
+deny contains result if {
+	some material in materials
+	not startswith(material.uri, "git+")
```

Comment:
I think we could support different syntax here, though probably we could hold off on that till someone tells us what other syntax exists. Then we could have this as rule data...

Reply:
Yeah. We should hold off on that until there's a requirement. FWIW, chains always adds the `git+` prefix: https://github.com/tektoncd/chains/blob/24dadcad3db7ed7ec82ad1bb25a34e9e38845c1f/pkg/chains/formats/intotoite6/attest/attest.go#L81-L87